### PR TITLE
fix: add `screenProps` to DrawerNavigatorItemsProps

### LINKED ISSUE #96: DrawerItemsProps should have screenProps
It seems that the type definition file in `@types/react-navigation` repository is no longer maintained and react-navigation using from its local one.

I submit [an issue in that repository](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/32777) a long time ago and mention all of the developers of this repository but I got no answer in there so I submit again in this repository.

In react-navigation 3.x, type of props object pass to `contentComponent` is `DrawerItemsProps`. also drawer component created by `createDrawerNavigator` must wrapped by an object created by `createAppContainer` which has a `screenProps` used to send user additional props to drawer menu component.
example:
```
const DrawerNav = createDrawerNavigator(
  {
    SOME_ROUTE_NAME: {
      screen: <SomeSCreen/>
    }
  },
  {
    contentComponent: renderSidebar,
    initialRouteName: SOME_ROUTE_NAME,
  });

const AppContainer = createAppContainer(DrawerNav);

function renderSidebar(props: DrawerItemsProps) {
  const {props1, props2,} = props.screenProps; // <----- getting sreenProps, typescript error TS2339 here
  return (
    <Sidebar
      props1
      props2
    />
  );
}

.
.
.

render() {
    const screenProps = {
      props1: some_value1,
      props2: some_value2
    };

    return (
      <View>
        <AppContainer key={'app navigator'} screenProps={screenProps}/> // <------- passing screenProps using `NavigationContainerProps`
        ...
      </View>
    )
  }
```

### PR DESCRIPTION
**Pull request by SaeedZhiany**
fixes #96

### CODE REVIEW

**Comment by satya164** [comment]
Thanks for the PR. Can you rebase it against master so CI passes?

**Comment by SaeedZhiany** [comment]
It's done.

**Comment by SaeedZhiany** [comment]
@satya164 
I did rebase master. but I failed again. the massage says that `ScreenProps` does not exist.
previously there is a `ScreenProps` type like below:

```
export type ScreenProps = {
    [key: string]: any;
  };
```
 
is it renamed to another thing?

**Comment by SaeedZhiany** [comment]
@satya164 
I did rebase master. but It's failed again. the massage says that `ScreenProps` does not exist.
previously there is a `ScreenProps` type like below:

```
export type ScreenProps = {
    [key: string]: any;
  };
```
 
is it renamed to another thing?

**Comment by SaeedZhiany** [comment]
Are you mean I must change the line to line below?

`screenProps?: unknown`

File: src/types.tsx [hunk 1]
```diff
@@ -95,6 +95,7 @@ export type DrawerNavigatorItemsProps = {
   inactiveLabelStyle?: StyleProp<TextStyle>;
   iconContainerStyle?: StyleProp<ViewStyle>;
   drawerPosition: 'left' | 'right';
+  screenProps?: ScreenProps;
```
Comment (satya164) [inline src/types.tsx]:
```suggestion
  screenProps: unknown;
```

**Comment by satya164** [comment]
> where unkonwn type is defined? I can't find it.

It's a built-in type in TypeScript

File: src/types.tsx (hunk 1 above)
Comment (SaeedZhiany) [reply src/types.tsx]: shouldn't it be optional?

**Pull request merged**